### PR TITLE
release-24.3: ui, apiutil, server: surface multiple indexes within a range in the hot ranges page

### PR DESCRIPTION
**Pull request by angles-n-daemons**
Contained with the three linked PRs are the changes required to surface all of the contents within a range in the hot ranges DB console page.

Backport:
  * 1/1 commits from "sql: using catalog reader, allow descriptor scanning by span" (#133190)
  * 1/1 commits from "apiutil, roachpb: create utilities to map descriptors to ranges" (#133840)
  * 2/2 commits from "ui, server: modify hot ranges api and table to use new contents approx." (#134106)

Please see individual PRs for details.

/cc @cockroachdb/release

Fixes: #130997

Release justification: fixes an important bug within the db console

### CODE REVIEW

**Comment by dhartunian** [comment]
@angles-n-daemons can you explain why the same commit appears twice in this PR? It looks like the UI PR contained a slightly different version of a commit that was already merged. That's a bit confusing here.

**Comment by angles-n-daemons** [comment]
I was unable to reproduce the duplicate commit locally. Looking into the individual commits themselves, the second one seems to add in some code which was removed in an amend. It could be a strange rebasing issue where some information was removed, which created changed the commit references.

File: pkg/server/serverpb/status.proto [hunk 1]
```diff
@@ -1484,11 +1484,11 @@ message HotRangesResponseV2 {
...
     ];
-    // table_name indicates the SQL table that the range belongs to.
-    string table_name = 4;
-    // database_name indicates on database that has current hot range.
+    // table_name has been deprecated in favor of tables = 16;
+    string table_name = 4 [deprecated = true];
+    // database_name has been deprecated in favor of databases = 17;
     string database_name = 5;
-    // index_name indicates the index name for current range.
+    // index_name has been deprecated in favor of indexes = 17;
```
Comment (dhartunian) [inline pkg/server/serverpb/status.proto:1491]: is there a reason that only one field has `deprecated = true`?

**Review by dhartunian — APPROVED** [review]

File: pkg/server/serverpb/status.proto (hunk 1 above)
Comment (angles-n-daemons) [reply pkg/server/serverpb/status.proto:1491]: No this was an accident. I'll make the corresponding change to master.

**Pull request merged**